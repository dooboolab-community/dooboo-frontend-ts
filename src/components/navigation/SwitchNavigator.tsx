import { BrowserRouter, Route, Switch } from 'react-router-dom';
import React, { ReactElement } from 'react';

import Intro from '../screen/Intro';
import Temp from '../screen/Temp';

function SwitchNavigator(): ReactElement {
  return (
    <BrowserRouter>
      <div style={{ textAlign: 'center' }}>
        <Switch>
          <Route exact={true} path="/" render={(): ReactElement => <Intro />} />
          <Route render={(): ReactElement => <Temp />} />
        </Switch>
      </div>
    </BrowserRouter>
  );
}

export default SwitchNavigator;
