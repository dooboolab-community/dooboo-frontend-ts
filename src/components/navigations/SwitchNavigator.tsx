import {BrowserRouter, Route, Routes} from 'react-router-dom';

import Intro from '../pages/Intro';
import {ReactElement} from 'react';
import Temp from '../pages/Temp';

function SwitchNavigator(): ReactElement {
  return (
    <BrowserRouter>
      <div
        style={{
          textAlign: 'center',
          flex: 1,
          width: '100vw',
        }}
      >
        <Routes>
          <Route path="/" element={<Intro />} />
          <Route path="*" element={<Temp />} />
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default SwitchNavigator;
