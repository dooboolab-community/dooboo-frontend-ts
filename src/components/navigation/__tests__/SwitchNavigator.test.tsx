import { RenderResult, render } from '@testing-library/react';

import React from 'react';
import SwitchNavigator from '../SwitchNavigator';
import { createTestElement } from '../../../../test/testUtils';

const props = {};
const component = createTestElement(<SwitchNavigator {...props} />);

describe('[SwitchNavigator] rendering', () => {
  let testingLib: RenderResult;

  beforeEach(() => {
    testingLib = render(component);
  });

  it('should renders without crashing', () => {
    const { baseElement } = testingLib;
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});
