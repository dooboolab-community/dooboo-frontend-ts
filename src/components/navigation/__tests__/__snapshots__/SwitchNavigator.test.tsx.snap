// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SwitchNavigator] rendering should renders without crashing 1`] = `
<body>
  <div>
    <div
      style="text-align: center;"
    >
      <div
        class="sc-EHOje jnQdsg"
      >
        <div
          class="sc-bZQynM iahEzA"
        >
          <span
            class="sc-htoDjs bfpTlg"
          />
          <span
            class="sc-htoDjs bfpTlg"
          />
          <span
            class="sc-htoDjs bfpTlg"
          />
        </div>
        <div
          class="sc-gzVnrw gqpNDZ"
        >
          <button
            class="sc-bdVaJa hCovpY"
            style="display: flex; height: 60px;"
          >
            <div
              class="sc-bwzfXH jccYBR"
            >
              <img
                class="sc-bxivhb aUweh"
                src="test-file-stub"
              />
              <span
                class="sc-htpNat fsEAzg"
              >
                Login
              </span>
            </div>
          </button>
          <button
            class="sc-bdVaJa hCovpY"
            style="display: flex; height: 60px;"
          >
            <div
              class="sc-bwzfXH jccYBR"
            >
              <span
                class="sc-htpNat fsEAzg"
              >
                Navigate
              </span>
            </div>
          </button>
          <button
            class="sc-bdVaJa hCovpY"
            style="display: flex; height: 60px;"
          >
            <div
              class="sc-bwzfXH jccYBR"
            >
              <span
                class="sc-htpNat fsEAzg"
              >
                Change theme
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
